File: app/src/main/java/com/example/roomshare/data/Bill.java
package com.example.roomshare.data;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(
    tableName = "bills",
    foreignKeys = @ForeignKey(
        entity = RoomEntity.class,
        parentColumns = "id",
        childColumns = "room_id",
        onDelete = ForeignKey.CASCADE
    ),
    indices = {@Index(value = "room_id")}
)
public class Bill {
    @PrimaryKey(autoGenerate = true)
    public long id;
    
    public String name;
    
    public double amount;
    
    @ColumnInfo(name = "room_id")
    public long roomId;

    public Bill(long id, String name, double amount, long roomId) {
        this.id = id;
        this.name = name;
        this.amount = amount;
        this.roomId = roomId;
    }

    @Ignore
    public Bill(String name, double amount, long roomId) {
        this.id = 0;
        this.name = name;
        this.amount = amount;
        this.roomId = roomId;
    }
}

File: app/src/main/java/com/example/roomshare/data/Bill.kt
package com.example.roomshare.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "bills",
    foreignKeys = [ForeignKey(
        entity = RoomEntity::class,
        parentColumns = ["id"],
        childColumns = ["room_id"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index(value = ["room_id"])]
)
data class Bill(
    @PrimaryKey(autoGenerate = true)
    val id: Long,
    val name: String,
    val amount: Double,
    @ColumnInfo(name = "room_id")
    val room_id: Long
) {
    constructor(name: String, amount: Double, room_id: Long) : 
        this(0, name, amount, room_id)
}

